### PR TITLE
[mds-transaction-api] [mds-transaction-service] Add Transaction API Tests & Global Error Handling Middleware

### PR DESCRIPTION
**Pull request by avatarneil**
## 📚 Purpose
This PR adds some rudimentary tests to the Transaction API, as well as adding a global error handling middlware to be used by the entire API. Got to play around with some custom `expect` matchers here for URL matching, which was fun! These aren't incredibly extensive tests by any means, but they do flex the main point of complexity at the moment, which is the GET `/transactions` querying. More tests to come in later PRs, I think. I decided to break a few things in the service tests out into fixtures so they could be used by the API tests as well.

## 📦 Impacts:
- mds-transaction-api
- mds-transaction-service

### CODE REVIEW

File: packages/mds-transactions-api/middleware/transaction-api-error.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  error: Error,
+  req: TransactionApiRequest,
+  res: TransactionApiResponse,
+  next: express.NextFunction
+) => {
+  log.error(req.method, req.originalUrl, error)
+
+  if (isError(error, ValidationError)) return res.status(HttpStatus.BAD_REQUEST).send({ error })
+
+  if (isError(error, BadParamsError)) return res.status(HttpStatus.BAD_REQUEST).send({ error })
```
Comment (cjlynch12) [inline packages/mds-transactions-api/middleware/transaction-api-error.ts:19]: Just confirming, both of these first two cases are `BAD_REQUEST` responses?
If so, does it make sense to consolidate into one case?
``` if (isError(error, ValidationError) || isError(error, BadParamsError)) do stuff```

File: packages/mds-transactions-api/tests/transaction-operations.spec.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import supertest from 'supertest'
+import { ApiServer } from '@mds-core/mds-api-server'
+import { transactionOperationsGenerator, TransactionServiceClient } from '@mds-core/mds-transactions-service'
+// import { SCOPED_AUTH } from '@mds-core/mds-test-data'
```
Comment (cjlynch12) [inline packages/mds-transactions-api/tests/transaction-operations.spec.ts:20]: remove if not needed

File: packages/mds-transactions-api/tests/transaction-statuses.spec.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import supertest from 'supertest'
+import { ApiServer } from '@mds-core/mds-api-server'
+import { transactionStatusesGenerator, TransactionServiceClient } from '@mds-core/mds-transactions-service'
+// import { SCOPED_AUTH } from '@mds-core/mds-test-data'
```
Comment (cjlynch12) [inline packages/mds-transactions-api/tests/transaction-statuses.spec.ts:20]: same, remove if not needed

File: packages/mds-transactions-api/tests/transaction-operations.spec.ts (hunk 2 above)
Comment (avatarneil) [reply packages/mds-transactions-api/tests/transaction-operations.spec.ts:20]: Reasoning below for why it's commented out at the moment... There's something very odd with importing `mds-test-data` from a Jest test suite within the monorepo that causes an OOM runtime error 😢

Comment (cjlynch12) [reply packages/mds-transactions-api/tests/transaction-operations.spec.ts:20]: big oof, but 👍

File: packages/mds-transactions-api/middleware/transaction-api-error.ts (hunk 1 above)
Comment (avatarneil) [reply packages/mds-transactions-api/middleware/transaction-api-error.ts:19]: Yup, should both be BAD_REQUEST responses. Could definitely consolidate into one case; it used to be more onerous prior to `isError` existing, but should be fine now

**Review by cjlynch12 — APPROVED** [review]
Yay, tests!

File: packages/mds-transactions-api/middleware/transaction-api-error.ts (hunk 1 above)
Comment (avatarneil) [reply packages/mds-transactions-api/middleware/transaction-api-error.ts]: Resolved in 33296d3

**Review by nessur — APPROVED** [review]
Looks great! love the middleware error handling

**Pull request merged**